### PR TITLE
Add item ratio

### PR DESCRIPTION
**Pull request by shaibv**
### What changed

...

### Why it changed

...

---

- [ ] Change is tested
- [ ] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

File: src/AddItem/AddItem.spec.js [hunk 1]
```diff
@@ -39,24 +37,33 @@ describe('AddItem', () => {
...
-      expect(driver.getHeight()).toEqual('400px');
-      expect(driver.getWidth()).toEqual('300px');
+      expect(driver.getContainerClasses()).toContain('ratio16x9');
     });
 
-    it('should not add style attribute when width and height props are not passed', () => {
-      props = {};
+    it('should have default asspect ratio 1x1', () => {
+      driver = createDriver(<AddItem/>);
+      expect(driver.getContainerClasses()).toContain('ratio1x1');
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: getContainerClasses is too low level. Maybe `getRatio` method?

File: src/AddItem/AddItem.spec.js [hunk 2]
```diff
@@ -39,24 +37,33 @@ describe('AddItem', () => {
 
   describe('height and width', () => {
 
-    it('should be added to style attribute when item is not present', () => {
+    it('should be use asspect ratio from props', () => {
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: remove the "be"

File: src/AddItem/AddItem.spec.js [hunk 3]
```diff
@@ -39,24 +37,33 @@ describe('AddItem', () => {
...
     });
 
-    it('should not add style attribute when width and height props are not passed', () => {
-      props = {};
+    it('should have default asspect ratio 1x1', () => {
+      driver = createDriver(<AddItem/>);
+      expect(driver.getContainerClasses()).toContain('ratio1x1');
+    });
+
+    it('should be ignore asspect ratio from props when height is given', () => {
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: remove the "be"

File: src/AddItem/AddItem.spec.js [hunk 4]
```diff
@@ -39,24 +37,33 @@ describe('AddItem', () => {
...
-      expect(driver.getWidth()).toEqual('');
+      expect(driver.getHeight()).toEqual('300px');
     });
 
   });
 
   describe('hide or show add item', () => {
 
-    it('should have an Add item tooltip', () => {
+    it('should have an tooltip with given content', () => {
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: "a tooltip"

File: src/Mixins/layout.scss [hunk 5]
```diff
@@ -0,0 +1,16 @@
+@mixin aspect-ratio($width, $height) {
```
Comment (lbelinsk) [inline src/Mixins/layout.scss]: I would name the `mixin` folder with small letters since it is not a component folder

File: src/Mixins/layout.scss [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  position: relative;
+  &:before {
+    display: block;
+    content: "";
+    width: 100%;
+    padding-top: ($height / $width) * 100%;
```
Comment (lbelinsk) [inline src/Mixins/layout.scss]: not 100% sure, but don't we have to put `calc` here?

File: src/AddItem/AddItem.js [hunk 7]
```diff
@@ -24,33 +33,35 @@ class AddItem extends WixComponent {
...
+      aspectRatio
     } = this.props;
-
+    const ratio = !height && ratioClasses[aspectRatio] ? ratioClasses[aspectRatio] : ratioClasses[defaultRatio];
     return (
-      <div className={style.container} style={{width, height}} data-hook="add-container">
-        <div data-hook="add-area" className={style.addLogo} onClick={onClick}>
-          {
+      <div className={classNames(ratio, style.box)} style={{height}} >
+        <div className={style.container} data-hook="add-container">
```
Comment (lbelinsk) [inline src/AddItem/AddItem.js]: I think we can have 2 divs, no need for 3

File: src/Mixins/layout.scss (hunk 5 above)
Comment (lbelinsk) [reply src/Mixins/layout.scss]: Maybe github didn't understand that you changed the file name?

File: src/AddItem/AddItem.spec.js [hunk 8]
```diff
@@ -39,24 +37,33 @@ describe('AddItem', () => {
...
 
-    it('should be added to style attribute when item is not present', () => {
+    it('should use asspect ratio from props', () => {
+      props = {
+        aspectRatio: '16/9'
+      };
       driver = createDriver(<AddItem {...props}/>);
-      expect(driver.getHeight()).toEqual('400px');
-      expect(driver.getWidth()).toEqual('300px');
+      expect(driver.getRadio()).toEqual('ratio16x9');
```
Comment (lbelinsk) [inline src/AddItem/AddItem.spec.js]: `getRadio` --> `getRatio`
And I think it will be nicer just to return `16x9` not `ratio16x9` (think about the end user)

File: src/AddItem/AddItem.scss [hunk 9]
```diff
@@ -1,26 +1,43 @@
 @import '../common';
+@import '../Mixins/layout';
```
Comment (lbelinsk) [inline src/AddItem/AddItem.scss]: change to mixins, this is breaking the build

File: src/AddItem/AddItem.driver.js [hunk 10]
```diff
@@ -3,25 +3,29 @@ import tooltipDriverFactory from '../Tooltip/Tooltip.driver';
...
+  const ratioRegex = /ratio\d+x\d+/;
+  const byHook = dataHook => element.querySelector(`[data-hook="${dataHook}"]`);
+  const addButton = () => byHook('add-container');
   const addTooltip = () => byHook('add-tooltip');
 
   return {
     /** return the driver element */
     element: () => element,
+    /** return the element ratio class */
+    getRadio: () => {
```
Comment (lbelinsk) [inline src/AddItem/AddItem.driver.js]: rename to `getRatio`

**Comment by lbelinsk** [comment]
Looks good, I am approving, but 2 things:
1. Like I said, I really think that `onAdditem` is a confusing name. I think onClick is much more appropriate, since it is actually being called onClick and not after the user selected the image or whatever.
2. A UX review from @milkyfruit / @wuwa will be great - https://wix-wix-style-react-pr-1822.surge.sh/?selectedKind=3.%20Inputs&selectedStory=3.12%20AddItem&full=0&addons=0&stories=1&panelRight=0

**Comment by lbelinsk** [comment]
Oh, @shaibv I see you did changed to `onClick`, but you didn't update it in the story!

File: src/AddItem/AddItem.spec.js [hunk 11]
```diff
@@ -77,14 +84,6 @@ describe('AddItem', () => {
       expect(wrapper.find('Tooltip').exists()).toBeFalsy();
     });
 
-    // it('should not have an Add item tooltip', () => {
```
Comment (shlomitc) [inline src/AddItem/AddItem.spec.js]: why did you remove it?

**Comment by lbelinsk** [comment]
The diffs are better now, but still there is a slight difference with the "+" sign and with the right border when we have error icon
https://eyes.applitools.com/app/test-results/00000251873062686178/00000251873062467367/steps/1/edit?accountId=mE99eP8CUn9UTSXSv98L3OSaJ0NZmAcY6Wndx6WSHODdY110

File: src/AddItem/AddItem.scss [hunk 12]
```diff
@@ -38,12 +55,5 @@
   position: absolute;
```
[the remark this answers is not in the document]
Comment (shaibv) [reply src/AddItem/AddItem.scss]: Good Catch!

File: src/AddItem/AddItem.spec.js (hunk 11 above)
Comment (shaibv) [reply src/AddItem/AddItem.spec.js]: It's tested in test: "should not have an AddItem tooltip markup"

Comment (shlomitc) [reply src/AddItem/AddItem.spec.js]: so let's delete it

Comment (shlomitc) [reply src/AddItem/AddItem.spec.js]: but this is this test and it is commented out

Comment (shlomitc) [reply src/AddItem/AddItem.spec.js]: ok, got it. it's in the previous tests

**Comment by lbelinsk** [comment]
Fixed in this pr: https://github.com/wix/wix-style-react/pull/1839
